crates/hdk3/src/host_fn/sign.rs

```diff
@@ -1,8 +1,23 @@
 use crate::prelude::*;
 
-/// Sign some data using the private key for the passed public key
+/// Sign something that is serializable using the private key for the passed public key.
+pub fn sign<K, D>(key: K, data: D) -> ExternResult<Signature>
+where
+    K: Into<AgentPubKey>,
+    D: serde::Serialize + std::fmt::Debug,
+{
+    host_call::<Sign, Signature>(__sign, Sign::new(key.into(), data)?)
+}
+
+/// Sign some data using the private key for the passed public key.
+///
+/// Assuming the private key for the provided pubkey exists in lair this will work.
+/// If we don't have the private key for the public key then we can't sign anything!
 ///
-/// Assuming the private key for the provided
-pub fn sign(key: AgentPubKey, data: Vec<u8>) -> ExternResult<Signature> {
-    host_call::<Sign, Signature>(__sign, Sign { key, data })
+/// @see sign
```

Comment:
I can't find this syntax in the intra-doc links anywhere. are you sure its understood by rustodcs?

Reply:
From the docs at https://doc.rust-lang.org/rustdoc/linking-to-items-by-name.html this would work like this:

```suggestion
/// See [`sign`](fn@sign)
```